app/account/edit/components/account-form.tsx

```diff
@@ -0,0 +1,164 @@
+import { useState } from 'react'
+import { useForm, Controller } from 'react-hook-form'
+import {
+  Box,
+  FormControl,
+  FormLabel,
+  FormErrorMessage,
+  Image,
+  VStack,
+  HStack,
+  Flex,
+  Input
+} from '@chakra-ui/react'
+
+import { MdAccountCircle } from 'react-icons/md'
+import { AccountSchema, accountResolver } from '@/account/schema'
+import { PrimaryButton, SecondaryButton } from '@/components/button'
+import { InputForm } from '@/components/input'
+import { Loading } from '@/components/loading'
+import { useUserId } from '@/providers/session-provider'
+import { useUploadFile, useUserUpdate } from '../../hooks'
+
+export type UserDetailsProps = {
+  name: string
+  email: string
+  profile_picture_url: string | undefined | null
+}
+
+export const AccountEditForm = ({
+  userDetails
+}: {
+  userDetails: UserDetailsProps
+}) => {
+  const userId = useUserId()
+  const { uploadFile } = useUploadFile()
+  const { updateUser, isUserUpdating, isUserUpdatingSuccessfull } =
+    useUserUpdate()
+  const [selectedImage, setSelectedImage] = useState<File | null>(null)
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors }
+  } = useForm<AccountSchema>({
+    defaultValues: {
+      name: userDetails?.name,
+      email: userDetails?.email,
+      profile_picture_url: userDetails?.profile_picture_url,
+      uploaded_image_file: null
+    },
+    resolver: accountResolver
+  })
+
+  const createHandler = handleSubmit(async (data: AccountSchema) => {
+    const { name, email } = data
+
+    if (name !== undefined && email !== undefined) {
+      if (selectedImage && userDetails?.profile_picture_url) {
+        await uploadFile(userDetails.profile_picture_url, selectedImage, userId)
+      }
+
+      updateUser(name, email)
+    }
+  })
+
+  return (
+    <>
+      {!isUserUpdatingSuccessfull && !isUserUpdating && (
```

Comment:
Question! Why you need this condition? If this is to avoid to display the form after updating user account, you can delete it because we don't want to display the form at all even empty. We need to fix in a different way.